exp/api/remote/remote_api.go

```diff
@@ -448,6 +448,14 @@ func WithWriteHandlerMiddlewares(middlewares ...func(http.Handler) http.Handler)
 	}
 }
 
+// maxDecodedSize limits the maximum allowed size of decompressed snappy payloads.
+// This protects against maliciously crafted payloads that could cause excessive memory
+// allocation and potentially lead to out-of-memory (OOM) conditions.
+// All usual payloads should be much smaller than this limit and pass without any problems.
+//
+// See more in https://github.com/prometheus/client_golang/pull/1917
```

Comment:
I think this comment is clear enough, perhaps no need for anyone to dig old PR here. It's a great change and clear enough to understand 👍🏽 

```suggestion
```